src/ansys/dpf/post/simulation.py

```diff
@@ -0,0 +1,524 @@
+                raise ValueError(
+                    "Argument 'components' must be an int, a str, or a list of either."
+                )
+            for comp in components:
+                if not (isinstance(comp, str) or isinstance(comp, int)):
+                    raise ValueError(
+                        "Argument 'components' can only contain integers and/or strings."
+                    )
+                if str(comp) not in self._principal_names:
+                    raise ValueError(
+                        "A principal component ID must be one of: "
+                        f"{self._principal_names}."
+                    )
+                out.append(int(comp) - 1)
+
+        # Take unique values
+        if out is not None:
+            out = list(set(out))
+        # Build columns names
+        if out is None:
+            columns = [base_name + str(comp) for comp in self._principal_names]
+        else:
+            columns = [base_name + self._principal_names[i] for i in out]
+        return out, columns
+
+    def _build_result_operator(
+        self,
+        name: str,
+        location: Union[locations, str],
+        force_elemental_nodal: bool,
+    ) -> dpf.Operator:
+        op = self._model.operator(name=name)
+        op.connect(7, self.mesh._meshed_region)
+        if force_elemental_nodal:
+            op.connect(9, "ElementalNodal")
+        else:
+            op.connect(9, location)
+        return op
+
+
+class MechanicalSimulation(Simulation, ABC):
+    """Base class for mechanical type simulations.
+
+    This class provides common methods and properties for all mechanical type simulations.
+    """
+
+    def __init__(self, result_file: PathLike):
+        """Instantiate a mechanical type simulation."""
+        model = dpf.Model(result_file)
+        data_sources = model.metadata.data_sources
+        super().__init__(data_sources=data_sources, model=model)
+
+    def _build_selection(
+        self,
+        selection: Union[Selection, None],
+        set_ids: Union[int, List[int], None],
+        times: Union[float, List[float], None],
+        load_steps: Union[
+            int, List[int], Tuple[int, Union[int, List[int]]], None
+        ] = None,
+        all_sets: bool = False,
+        named_selections: Union[List[str], str, None] = None,
+        element_ids: Union[List[int], None] = None,
+        node_ids: Union[List[int], None] = None,
+        location: Union[locations, str] = locations.nodal,
+    ) -> Selection:
+        tot = (
+            (node_ids is not None)
+            + (element_ids is not None)
+            + (named_selections is not None)
+            + (selection is not None)
+        )
+        if tot > 1:
+            raise ValueError(
+                "Arguments selection, named_selections, element_ids, "
+                "and node_ids are mutually exclusive"
+            )
+        if selection is not None:
+            return selection
```

Comment:
why you want this??

Like if you pass ``Selection``... it retuns selection without doing anything....